### PR TITLE
pylint: 2.17.5 -> 3.0.1, python311Packages.astroid: 2.15.6 -> 3.0.1, nbqa: 1.7.0 -> 1.7.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelogs:
- python311Packages.astroid: https://github.com/PyCQA/astroid/blob/refs/tags/v3.0.1/ChangeLog
- pylint: https://github.com/pylint-dev/pylint/releases/tag/v3.0.1
- nbqa: https://github.com/nbQA-dev/nbQA/compare/1.7.0...1.7.1

cc @l0b0 @geluk 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by l0b0 — APPROVED** [review]

**Comment by natsukium** [comment]
This will cause a mass rebuild, so please change the target to staging.

**Comment by GaetanLepage** [comment]
> This will cause a mass rebuild, so please change the target to staging.

You are right. I didn't expect that many rebuilds.
If I rebase my branch on staging and force push, this will ping a massive amount of person right ?
Do I need to open a separate PR and close this one ?

**Comment by natsukium** [comment]
Following [this](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging), it will probably be fine.
If you are concerned about mass pinging, you can make a new PR.

**Comment by GaetanLepage** [comment]
> Following [this](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging), it will probably be fine.

Thanks ! It looks like it worked :)

File: pkgs/development/python-modules/astroid/default.nix [hunk 1]
```diff
@@ -16,16 +16,16 @@
 
 buildPythonPackage rec {
   pname = "astroid";
-  version = "2.15.6"; # Check whether the version is compatible with pylint
-  format = "pyproject";
+  version = "3.0.1"; # Check whether the version is compatible with pylint
```
Comment (l0b0) [inline pkgs/development/python-modules/astroid/default.nix:19]: Does that comment still belong there?

**Review by l0b0 — APPROVED** [review]

File: pkgs/development/python-modules/astroid/default.nix [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
   src = fetchFromGitHub {
     owner = "PyCQA";
-    repo = pname;
+    repo = "astroid";
     rev = "refs/tags/v${version}";
-    hash = "sha256-0oNNEVD8rYGkM11nGUD+XMwE7xgk7mJIaplrAXaECFg=";
+    hash = "sha256-z6FmB3I3VmiIx0MSsrkvHmDVv2h3CaaeXlDG3DewGXw=";
   };
 
   nativeBuildInputs = [
```
Comment (mweinelt) [inline pkgs/development/python-modules/astroid/default.nix:31]: Drop `wheel` here. It isn't required.

File: pkgs/development/python-modules/astroid/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/astroid/default.nix:1]: Clean up the propagatedBuildInputs and remove duplicate packages from nativeCheckInputs.

https://github.com/pylint-dev/astroid/blob/v3.0.1/pyproject.toml#L32-L34

Comment (mweinelt) [inline pkgs/development/python-modules/astroid/default.nix:1]: Check if the disabled test can be dropped.

File: pkgs/development/python-modules/pylint/default.nix [hunk 3]
```diff
@@ -24,32 +23,18 @@
...
-    (fetchpatch {
-      name = "unpin-setuptools.patch";
-      url = "https://github.com/pylint-dev/pylint/commit/a0ac282d6f8df381cc04adc0a753bec66fc4db63.patch";
-      hash = "sha256-15O72LE2WQK590htNc3jghdbVoGLHUIngERDpqT8pK8=";
-    })
-  ];
-
   nativeBuildInputs = [
     setuptools
     wheel
```
Comment (mweinelt) [inline pkgs/development/python-modules/pylint/default.nix:40]:
```suggestion
```

File: pkgs/development/python-modules/pylint/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pylint/default.nix:1]: Add `meta.changelog`.

https://github.com/pylint-dev/pylint/releases/tag/v3.0.2

File: pkgs/tools/misc/nbqa/default.nix [hunk 4]
```diff
@@ -7,13 +7,13 @@
 }:
 python3.pkgs.buildPythonApplication rec {
   pname = "nbqa";
-  version = "1.7.0";
+  version = "1.7.1";
```
Comment (mweinelt) [inline pkgs/tools/misc/nbqa/default.nix:10]: Enable the PEP517 builder.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/astroid/default.nix [hunk 5]
```diff
@@ -16,41 +16,34 @@
...
   nativeCheckInputs = [
     pip
     pytestCheckHook
-    typing-extensions
   ];
 
   disabledTests = [
     # DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('tests.testdata.python3.data.path_pkg_resources_1.package')`.
-    "test_identify_old_namespace_package_protocol"
+    # "test_identify_old_namespace_package_protocol"
```
Comment (mweinelt) [inline pkgs/development/python-modules/astroid/default.nix:46]: Will not merge like that.

Comment (mweinelt) [reply pkgs/development/python-modules/astroid/default.nix:46]: My recommendation would be to test it locally while rebased against master.

Comment (mweinelt) [reply pkgs/development/python-modules/astroid/default.nix]: Builds for me.

**Pull request merged**